api/chartRepo/ChartRepositoryRestHandler.go

```diff
@@ -283,4 +304,69 @@ func (handler *ChartRepositoryRestHandlerImpl) DeleteChartRepo(w http.ResponseWr
 		return
 	}
 	common.WriteJsonResp(w, nil, CHART_REPO_DELETE_SUCCESS_RESP, http.StatusOK)
-}
\ No newline at end of file
+}
+
+func (handler *ChartRepositoryRestHandlerImpl) CreateChartFromBinary(w http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+	var bean ChartBinary
+	err := decoder.Decode(&bean)
+	if err != nil {
+		handler.Logger.Errorw("request err, HandleGitWebhook", "err", err, "payload", bean)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+
+	binaryfile, err := base64.StdEncoding.DecodeString(bean.BinaryFile)
+
+	err = util2.ExtractTarGz(bytes.NewReader(binaryfile), string(handler.refChartDir))
+	if err != nil {
+		fmt.Println("Error Retrieving the File")
+		common.WriteJsonResp(w, err, "Error Retrieving the File", http.StatusBadRequest)
+		return
+	}
+
+	chartName := bean.BinaryFileName
+	chartVersion := bean.Version
+
+	var chartLocation string
+	if !strings.Contains(chartName, strings.ReplaceAll(chartVersion, ".", "-")) {
+		chartLocation = chartName + "_" + strings.ReplaceAll(chartVersion, ".", "-")
+	} else {
+		chartLocation = chartName
+	}
+
+	charts, err := handler.chartRefRepository.GetAll()
```

Comment:
fetch charts with gz data as it is not required for comparison and ll increase memory consumption